packages/builder/test/helpers/environment.ts

```diff
@@ -1,18 +1,25 @@
 import { resetBuilderContext } from "../../src/internal/reset";
-import { AlgobRuntimeEnv } from "../../src/types";
+import { AlgobRuntimeEnv, PromiseAny } from "../../src/types";
 
 declare module "mocha" {
   interface Context {
     env: AlgobRuntimeEnv;
   }
 }
 
-export function useEnvironment() {
-  beforeEach("Load environment", function () {
+export function useEnvironmentWithBeforeEach(beforeEachFn?: (algobRuntimeEnv: AlgobRuntimeEnv) => PromiseAny) {
+  beforeEach("Load environment", async function () {
     this.env = require("../../src/internal/lib/lib");
+    if (beforeEachFn) {
+      await beforeEachFn(this.env)
+    }
```

Comment:
here as well, async - await is not needed. Just `return beforeEachFn(...)` - it's a promise already.